### PR TITLE
Clean up NCursesW finder

### PR DESCRIPTION
**Pull request by ShadowNinja**
Previously I had to specify `-DCURSES_INCLUDE_PATH=/usr/include` and
`-DCURSES_HAVE_NCURSES_H=TRUE` to get ncurses to work, when those were obvious defaults.  Now they're automatically found.

I also moved the header decision logic from the source to the CMake module.

### CODE REVIEW

**Comment by sofar** [comment]
And a lot less work as well, nice.

File: cmake/Modules/FindNcursesw.cmake [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-# FindNcursesw
-# ------------
-#
-# Find the ncursesw (wide ncurses) include file and library.
+# Find an ncurses include file and library.
 #
 # Based on FindCurses.cmake which comes with CMake.
 #
-# Checks for ncursesw first. If not found, it then executes the
+# Checks for ncursesw first.  If not found, it then executes the
```
Comment (est31) [inline cmake/Modules/FindNcursesw.cmake]: I don't think this is required.

File: cmake/Modules/FindNcursesw.cmake [hunk 2]
```diff
@@ -70,120 +55,74 @@
...
-#
-# The above copyright and license notice applies to distributions of
-# CMake in source and binary form.  Some source files contain additional
-# notices of original copyright by their contributors; see each source
-# for details.  Third-party software packages supplied with CMake under
-# compatible licenses provide their own copyright notices documented in
-# corresponding subdirectories.
-#
-# ------------------------------------------------------------------------------
-#
```
Comment (est31) [inline cmake/Modules/FindNcursesw.cmake]: Removing this is rude IMO. Cmake was developed with a number of contributors, and if this is the way they chose attribution we should respect it, if we copy the file. Yes, its not required by the license of the file which only covers the direct copyright statement, but its more polite if we keep it in.

File: cmake/Modules/FindNcursesw.cmake [hunk 3]
```diff
@@ -70,120 +55,74 @@
...
-  CURSES_EXTRA_LIBRARY
-  CURSES_FORM_LIBRARY
-  )
+	CURSES_INCLUDE_PATH
+	CURSES_CURSES_LIBRARY
+	CURSES_NCURSESW_LIBRARY
+	CURSES_EXTRA_LIBRARY
+	CURSES_FORM_LIBRARY
+)
+
```
Comment (est31) [inline cmake/Modules/FindNcursesw.cmake]: This is a blank line at EOF, git am complains about it.

**Comment by est31** [comment]
Otherwise looks fine.

File: cmake/Modules/FindNcursesw.cmake (hunk 2 above)
Comment (ShadowNinja) [inline cmake/Modules/FindNcursesw.cmake]: I don't think we need to list a history of everyone that contributed to the file in it.  The copyright alone is already about half of the file.

File: cmake/Modules/FindNcursesw.cmake (hunk 1 above)
Comment (ShadowNinja) [inline cmake/Modules/FindNcursesw.cmake]: No, not strictly.

File: cmake/Modules/FindNcursesw.cmake (hunk 2 above)
Comment (sofar) [inline cmake/Modules/FindNcursesw.cmake]: I'd prefer not removing this type of attribution.

**Comment by est31** [comment]
This PR can go in if the two things are fixed.

File: cmake/Modules/FindNcursesw.cmake (hunk 3 above)
Comment (ShadowNinja) [inline cmake/Modules/FindNcursesw.cmake]: I believe we decided to add these blank lines a long time ago.

Comment (est31) [inline cmake/Modules/FindNcursesw.cmake]: Yes, one `\n` should be at the end of a file, but this is one extra blank line.

Comment (ShadowNinja) [inline cmake/Modules/FindNcursesw.cmake]: Actually, it's just one.

File: cmake/Modules/FindNcursesw.cmake (hunk 2 above)
Comment (ShadowNinja) [inline cmake/Modules/FindNcursesw.cmake]: I've only removed two things:
  * A notice about where the license applies in CMake, which is only relevant for the CMake source.
  * A sponsorship list for CMake, which is also irrelevant.

I've left the attribution information above, I just don't think that these sections provide any useful information.

**Comment by est31** [comment]
Hmm perhaps it would be good to have this inside 0.4.14, to make life easier for distro packagers?

**Comment by est31** [comment]
@paramat what do you think

**Comment by nerzhul** [comment]
@ShadowNinja can you rebase ?

**Comment by rubenwardy** [comment]
lgtm, but needs a rebase

**Comment by paramat** [comment]
@ShadowNinja Seems to have approval so please rebase.

**Comment by paramat** [comment]
@ShadowNinja if you have no time for this let us know so someone can adopt.

**Comment by paramat** [comment]
Needs adoption and rebase, already approved.

**Comment by paramat** [comment]
Needs adoption and rebase, already approved.

**Comment by paramat** [comment]
Closing due to neglect, can be reopened once rebased. Rebase looks quick and easy.

**Comment by paramat** [comment]
IRC http://irc.minetest.net/minetest-dev/2018-01-02#i_5186807

21:02 ShadowNinja paramat: The original bug was fixed in a different way (decbd39 and 27ee8d8), so it's mainly a cleanup now.  My version is much nicer, but this part of code is very platform-dependant, so I'd like some testing on various platforms to be sure it doesn't preak anything.
21:06 ShadowNinja Looks like that was the only commit, so a rebase would just involve choosing my version of the file.
21:09 ShadowNinja I'll rebase it.

**Comment by paramat** [comment]
Removed +1 by @rubenwardy as the situation has changed so much, hope that's ok?

**Pull request closed without merging**